src/Main.hs

```diff
@@ -37,10 +38,10 @@ opts = info (options <**> helper) $ fullDesc
 main :: IO ()
 main = do (Options f c cov conf) <- execParser opts
           cfg <- maybe (pure defaultConfig) parseConfig conf
-          flip runReaderT (cfg & cConf %~ (if cov then \k -> k {knownCoverage = Just mempty}
-                                                  else id)) $ do
-            (v,a,ts) <- loadSolidity f (pack <$> c)
-            let r = v ^. state . contract
-            let w = World (cfg ^. sConf . sender) [(r, a)]
-            let ts' = zip ts $ repeat r
-            ui v w ts' >> pure ()
+          Campaign r _ <- flip runReaderT (cfg & cConf %~ (if cov then \k -> k {knownCoverage = Just mempty}
+                                                                  else id)) $ do
+            (v, w, ts) <- loadTesting f (pack <$> c)
+            ui v w ts
+          if any (\case {Passed -> False; _ -> True;}) (snd <$> r)
```

Comment:
`Passed` doesn't have any args, so we can just derive `Eq` and do `== Passed`

Reply:
We can't use `deriving Eq` in `TestState`, because it uses `ExecException`, which uses `Error` (in HEVM), which uses a GADT: `Query` (in HEVM). :sob:

Reply:
Done. (@japesinator)